diff_drive_controller/src/diff_drive_controller_parameter.yaml

```
@@ -35,94 +39,112 @@ diff_drive_controller:
+    read_only: true,
   }
   tf_frame_prefix_enable: {
     type: bool,
     default_value: true,
     description:  "Enables or disables appending tf_prefix to tf frame id's.",
+    read_only: true,
   }
   tf_frame_prefix: {
     type: string,
     default_value: "",
     description:  "(optional) Prefix to be appended to the tf frames, will be added to odom_id and base_frame_id before publishing. If the parameter is empty, controller's namespace will be used.",
+    read_only: true,
   }
   odom_frame_id: {
     type: string,
     default_value: "odom",
     description:  "Name of the frame for odometry. This frame is parent of ``base_frame_id`` when controller publishes odometry.",
+    read_only: true,
   }
   base_frame_id: {
     type: string,
     default_value: "base_link",
     description: "Name of the robot's base frame that is child of the odometry frame.",
+    read_only: true,
   }
   pose_covariance_diagonal: {
     type: double_array,
     default_value: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
     description: "Odometry covariance for the encoder output of the robot for the pose. These values should be tuned to your robot's sample odometry data, but these values are a good place to start: ``[0.001, 0.001, 0.001, 0.001, 0.001, 0.01]``.",
+    read_only: true,
   }
   twist_covariance_diagonal: {
     type: double_array,
     default_value: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
     description: "Odometry covariance for the encoder output of the robot for the speed. These values should be tuned to your robot's sample odometry data, but these values are a good place to start: ``[0.001, 0.001, 0.001, 0.001, 0.001, 0.01]``.",
+    read_only: true,
   }
   open_loop: {
     type: bool,
     default_value: false,
     description: "If set to true the odometry of the robot will be calculated from the commanded values and not from feedback.",
+    read_only: false,
   }
   position_feedback: {
     type: bool,
     default_value: true,
     description: "Is there position feedback from hardware.",
+    read_only: false,
   }
   enable_odom_tf: {
     type: bool,
     default_value: true,
     description: "Publish transformation between ``odom_frame_id`` and ``base_frame_id``.",
+    read_only: false,
   }
   cmd_vel_timeout: {
     type: double,
     default_value: 0.5, # seconds
     description: "Timeout in seconds, after which input command on ``cmd_vel`` topic is considered staled.",
+    read_only: false,
   }
   publish_limited_velocity: {
     type: bool,
     default_value: false,
     description: "Publish limited velocity value.",
+    read_only: false,
   }
   velocity_rolling_window_size: {
     type: int,
     default_value: 10,
     description: "Size of the rolling window for calculation of mean velocity use in odometry.",
+    read_only: true,
   }
   publish_rate: {
     type: double,
     default_value: 50.0, # Hz
     description: "Publishing rate (Hz) of the odometry and TF messages.",
+    read_only: false,
```

```suggestion
    read_only: true,
```
This should be constant too